src/components/Modals/BuyItemSlotsModal/BuyItemSlotsModal.tsx

```diff
@@ -69,18 +80,28 @@ export default class BuyItemSlotsModal extends React.PureComponent<Props, State>
               <Loader active size="tiny" />
             ) : (
               <>
-                <Field label={t('buy_item_slots_modal.how_many_slots_title')} placeholder="1" onChange={this.handleSlotToBuyChange} />
+                <div className={hasError ? styles.errorField : ''}>
+                  <Field
+                    label={t('buy_item_slots_modal.how_many_slots_title')}
+                    placeholder="1"
+                    value={slotsToBuy}
+                    message={hasError ? t('buy_item_slots_modal.buy_slots_error') : undefined}
+                    error={hasError}
+                    onChange={this.handleSlotToBuyChange}
+                  />
+                </div>
                 <div className={styles.slotValue}>
                   {t('buy_item_slots_modal.slots_value', {
                     symbol: <Mana network={Network.MATIC} size="small" />,
                     slot_cost: slotPrice,
-                    total_cost: slotPrice && slotsToBuy ? slotPrice * Number(slotsToBuy) : 0
+                    total_cost: slotPrice && this.isValidSlotAmount() ? slotPrice * Number(slotsToBuy) : 0
                   })}
                 </div>
                 <div className={styles.slotValue}>
                   {t('buy_item_slots_modal.total_cost', {
                     symbol: <Mana network={Network.MATIC} size="small" />,
-                    total_cost: slotPrice && slotsToBuy ? applySlotBuySlippage(BigNumber.from(slotPrice).mul(slotsToBuy)).toString() : 0
+                    total_cost:
+                      slotPrice && this.isValidSlotAmount() ? applySlotBuySlippage(BigNumber.from(slotPrice).mul(slotsToBuy)).toString() : 0
```

Comment:
I think that if the user introduces something like `12.5`, this will try to apply the slippage to that number, since `slotPrice` is still the input from the user (we might need to call the `parseInt` when setting the state).
When the `parseInt` is called in the `isValidSlotAmount`, the `slotsAmount` will always be an integer, so I believe it will always return `true`, right?

Reply:
You're right, you also made me realize that we weren't checking against the available slots of the third party so I added that too

https://user-images.githubusercontent.com/692077/159593663-2f30240b-b5c5-408f-b751-2efa94b26299.mov

Reply:
Ohhh nice! looks awesome :) 
Changes approved!